definitions/scenarios/update.rb

```diff
@@ -0,0 +1,126 @@
+module Scenarios::Update
+  class Abstract < ForemanMaintain::Scenario
+    def self.update_metadata(&block)
+      metadata do
+        tags :update_scenario
+        instance_eval(&block)
+      end
+    end
+  end
+
+  class PreUpdateCheck < Abstract
+    update_metadata do
+      description 'Checks before updating'
+      tags :pre_update_checks
+      run_strategy :fail_slow
+    end
+
+    # rubocop:disable Metrics/MethodLength
+    def compose
+      add_steps(
+        Checks::Foreman::FactsNames, # if Foreman database present
+        Checks::ForemanProxy::CheckTftpStorage, # if Satellite with foreman-proxy+tftp
+        Checks::ForemanProxy::VerifyDhcpConfigSyntax, # if foreman-proxy+dhcp-isc
+        Checks::ForemanTasks::NotPaused, # if foreman-tasks present
+        Checks::Puppet::VerifyNoEmptyCacertRequests, # if puppetserver
+        Checks::ServerPing,
+        Checks::ServicesUp,
+        Checks::SystemRegistration,
+        Checks::CheckHotfixInstalled,
+        Checks::CheckTmout,
+        Checks::CheckUpstreamRepository,
+        Checks::Disk::AvailableSpace,
+        Checks::Disk::AvailableSpaceCandlepin, # if candlepin
+        Checks::Foreman::ValidateExternalDbVersion, # if external database
+        Checks::Foreman::CheckCorruptedRoles,
+        Checks::Foreman::CheckDuplicatePermissions,
+        Checks::Foreman::TuningRequirements, # if katello present
+        Checks::ForemanOpenscap::InvalidReportAssociations, # if foreman-openscap
+        Checks::ForemanTasks::Invalid::CheckOld, # if foreman-tasks
+        Checks::ForemanTasks::Invalid::CheckPendingState, # if foreman-tasks
+        Checks::ForemanTasks::Invalid::CheckPlanningState, # if foreman-tasks
+        Checks::ForemanTasks::NotRunning, # if foreman-tasks
+        Checks::NonRhPackages,
+        Checks::PackageManager::Dnf::ValidateDnfConfig,
```

Comment:
picking this particular, but we should double check this either way: not all those checks have sufficient "confines" -- like this one does not confine to the package manager actually being DNF and it might break on Debian-like systems.
(not that this would be different with the current upgrade commands, but I still wanted to point it out)

Reply:
Funny, this one doesn't throw a problem:

```
Check to validate dnf configuration before upgrade:                   [OK]
```

Not that we can't confine it better. I actually found other problems with running this on Debian that I can work on addressing separately.

Reply:
Funny, this one doesn't throw a problem:

```
Check to validate dnf configuration before upgrade:                   [OK]
```

Not that we can't confine it better. I actually found other problems with running this on Debian that I can work on addressing separately.

Reply:
Maybe it doesn't find any offending lines as the configuration is not there at all, I didn't look too closely, just that it's not confined.

Thanks!

Reply:
First item found to fix -- https://github.com/theforeman/foreman_maintain/pull/832